src/background/user_actions.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * 🎶 No alarms and no surprises 🎶
+ * 🎶 No alarms and no surprises 🎶
+ * https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/User_actions
+ */
+
+import * as excmds from "@src/.excmds_background.generated"
+import * as R from "ramda"
+import { messageTab } from "@src/lib/messaging"
+
+export async function escapehatch() {
+    const tabs = await browser.tabs.query({ currentWindow: true })
+    const tridactyl_tabs: browser.tabs.Tab[] = []
+    await Promise.all(
+        tabs.map(async tab => {
+            try {
+                // This doesn't actually return "true" like it is supposed to
+                await messageTab(tab.id, "alive")
+                tridactyl_tabs.push(tab)
+                return true
+            } catch (e) {
+                return false
+            }
+        }),
+    )
```

Comment:
You can also use Promise.allSettled to avoid needing to catch the `messageTab` call.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#Browser_compatibility it is a bit too new - not supported in FF68